commons/com.b2international.index/src/com/b2international/index/query/TextPredicate.java

```diff
@@ -72,11 +72,18 @@ public String toString() {
 
 	public TextPredicate withIgnoreStopwords(boolean ignoreStopwords) {
 		if (ignoreStopwords) {
-			analyzer = Analyzers.TOKENIZED_IGNORE_STOPWORDS.getAnalyzer();
+			return withAnalyzer((analyzer == Analyzers.TOKENIZED_SYNONYMS || analyzer == Analyzers.TOKENIZED_SYNONYMS_IGNORE_STOPWORDS) ? Analyzers.TOKENIZED_SYNONYMS_IGNORE_STOPWORDS : Analyzers.TOKENIZED_IGNORE_STOPWORDS);
 		} else {
-			analyzer = null;
+			return withAnalyzer(null);
```

Comment:
Why don't we have to do the reverse of the code block above? If the analyzer is currently set to `TOKENIZED(_SYNONYMS)?_IGNORE_STOPWORDS`, shouldn't it be changed back to `TOKENIZED$1` (with pseudo-regexp syntax)?

Reply:
Right 😄